### ISSUE TITLE
Crash on link click if text is selectable

### ISSUE THREAD

**Issue by pchupik** [Saketme/Better-Link-Movement-Method#11]
In sample app make the textView text-selectable (e.g. by adding `android:textIsSelectable="true"`)

1. Select some text in the textView which contains clickableSpan
2. Click on the clickableSpan
[crash.txt](https://github.com/Saketme/Better-Link-Movement-Method/files/1417766/crash.txt)
```
java.lang.IndexOutOfBoundsException: setSpan (-1 ... -1) starts before 0
...
at me.saket.bettermovementmethod.BetterLinkMovementMethod.highlightUrl(BetterLinkMovementMethod.java:342)
at me.saket.bettermovementmethod.BetterLinkMovementMethod.onTouchEvent(BetterLinkMovementMethod.java:222)
```

**Comment by Saketme** [comment]
Hey @pchupik, I'm busy traveling till 10th of Nov and so won't be able to look into this until then. :(

**Comment by pchupik** [comment]
Looks like I found a fix for this. In `BetterLinkMovementMethod.highlightUrl( )` add check textView for selection and clear focus if it has one:
```
text.setSpan(new BackgroundColorSpan(textView.getHighlightColor()), spanStart, spanEnd, Spannable.SPAN_INCLUSIVE_INCLUSIVE);
if (textView.hasSelection())
           textView.clearFocus();
textView.setText(text);
```

**Comment by Saketme** [comment]
Fixed in commit [d0a7db1](https://github.com/Saketme/Better-Link-Movement-Method/commit/d0a7db1f92a5e5cc2c165d98a5914bee37cef977). Will release a new version soon. In the meanwhile, you can get a build off HEAD using [jitpack.io](jitpack.io).

**Comment by Saketme** [comment]
Fixed in 2.1.0: https://github.com/Saketme/Better-Link-Movement-Method/releases/tag/v2.1.0
